File: Div3-Group1/ReplaceElementsInAnArray.cpp
class Solution {
public:
    vector<int> arrayChange(vector<int>& nums, vector<vector<int>>& operations) {
         
        unordered_map<int,int> m;
        
        for(int i=0; i<nums.size();i++) {
            m[nums[i]]=i;
        }
        for(auto a: operations)
        {
            int index=m[a[0]];
            nums[index]=a[1];
             m.erase(a[0]);
             m[a[1]]=index;
        }

    //     for (const auto& pair : m) {
    //     int key = pair.first;
    //     int value = pair.second;
    //     cout << "Key: " << key << ", Value: " << value << endl;
    // }

        return nums;
    }
};

File: leet-code-solutions/replace-elements-in-an-array.py
class Solution(object):
    def arrayChange(self, nums, operations):
        """
        :type nums: List[int]
        :type operations: List[List[int]]
        :rtype: List[int]
        """
        Dict={}

        for i in range(len(nums)):
            Dict[nums[i]]=i
        
        for operation in operations:
            item=operation[0]
            placeholder=operation[1]
            index=Dict[item]

            nums[index]=placeholder

            del Dict[item]
            
            Dict[placeholder]=index
        
        return nums 
        